### PR TITLE
Move post installation scripts to the end of queue for RHV-H

### PR DESCRIPTION
**Pull request by lveyde**
Resolves: rhbz#1953685

Signed-off-by: Lev Veyde <lveyde@redhat.com>

### CODE REVIEW

**Comment by lveyde** [comment]
@sandrobonazzola

**Comment by lveyde** [comment]
Related RHBZ: https://bugzilla.redhat.com/1953685

**Comment by lveyde** [comment]
> Hello @lveyde! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:
> 
> * In the file [`pyanaconda/installation.py`](https://github.com/rhinstaller/anaconda/blob/18453ed0a71c8d4015dc972c358bac30c1ac9c0a/pyanaconda/installation.py):
> 
> > [Line 182:100](https://github.com/rhinstaller/anaconda/blob/18453ed0a71c8d4015dc972c358bac30c1ac9c0a/pyanaconda/installation.py#L182): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 99 characters)
> > [Line 183:100](https://github.com/rhinstaller/anaconda/blob/18453ed0a71c8d4015dc972c358bac30c1ac9c0a/pyanaconda/installation.py#L183): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 99 characters)
> > [Line 211:100](https://github.com/rhinstaller/anaconda/blob/18453ed0a71c8d4015dc972c358bac30c1ac9c0a/pyanaconda/installation.py#L211): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 99 characters)
> > [Line 212:100](https://github.com/rhinstaller/anaconda/blob/18453ed0a71c8d4015dc972c358bac30c1ac9c0a/pyanaconda/installation.py#L212): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 99 characters)
> 
> ##### Comment last updated at 2021-04-27 18:00:50 UTC

Fixed

File: pyanaconda/installation.py [hunk 1]
```diff
@@ -176,9 +176,19 @@ def run_generate_initramfs():
...
+        post_scripts = TaskQueue(
+            "Post installation scripts",
+            N_("Running post-installation scripts")
+        )
+        post_scripts.append(Task(
+            "Run post installation scripts",
+            runPostScripts,
+            (ksdata.scripts,)
+        ))
+        configuration_queue.append(post_scripts)
```
Comment (sandrobonazzola) [inline pyanaconda/installation.py:191]: If there's no special reason to write the ks earlier, maybe we can just have it done later and not differentiate between RHVH and others.

**Review by poncovka — COMMENTED** [review]
I think that we could run `%post` scripts at the end of the installation by default. It won't have an effect on the generated kickstart file and it shouldn't cause any issues with the kexec setup. The documentation doesn't specify exactly when they should be run. What do you think, @jkonecny12, @jstodola?

**Comment by jstodola** [comment]
The more installation tasks are done before `%post`, the better. So it makes sense to do this change for all products, not just RHVH.

**Comment by jkonecny12** [comment]
I can't think of a way how to change kickstart data from `%post` section so I don't see a reason to have `%post` before that.

The only think that come to my mind is to change the output kickstart and attacking the future machines by this. However, I don't think this is really valid because if you can get to the `%post` here you can do much more harm.

**Comment by jkonecny12** [comment]
I can't think of a way how to change kickstart data from `%post` section so I don't see a reason to have `%post` before that.

The only think that come to my mind is to change the output kickstart and attacking the future machines by this. However, I don't think this is really valid because if you can get to the `%post` here you can do much more harm.

In other words, yes please move the `%post` at the end.

**Comment by poncovka** [comment]
@lveyde, could you please remove the checks for RHVH and run the `%post` scripts always at the end of the installation?

**Comment by lveyde** [comment]
> @lveyde, could you please remove the checks for RHVH and run the `%post` scripts always at the end of the installation?

Done

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

**Review by poncovka — APPROVED** [review]
Looks good to me. Thank you!

**Pull request merged**